File: Animation-OC/Animation-OC/ColorViewController.m

//
//  ColorViewController.m
//  Animation-OC
//
//  Created by chenyufeng on 15/12/31.
//  Copyright © 2015年 chenyufengweb. All rights reserved.
//

#import "ColorViewController.h"

@interface ColorViewController ()

@property (weak, nonatomic) IBOutlet UIView *greenSquare;

@end

@implementation ColorViewController

- (void)viewDidLoad {

  [super viewDidLoad];
}

- (void)viewWillAppear:(BOOL)animated{

  [super viewWillAppear:animated];
  [UIView animateWithDuration:2 animations:^{
    //宽高缩放比；
    self.greenSquare.backgroundColor = [UIColor brownColor];
  }];
}

@end

File: PositionAnimation01/PositionAnimation01/ColorViewController.swift
//
//  ColorViewController.swift
//  PositionAnimation01
//
//  Created by chenyufeng on 15/8/20.
//  Copyright (c) 2015年 chenyufengweb. All rights reserved.
//

import UIKit

class ColorViewController: UIViewController {

    @IBOutlet weak var greenSquare: UIView!
    @IBOutlet weak var swiftText: UILabel!
  
    override func viewDidLoad() {

        super.viewDidLoad()
    }

    override func viewDidAppear(animated: Bool) {

        super.viewDidAppear(animated)
        func anim(){

            self.greenSquare.backgroundColor = UIColor.blackColor()
            self.swiftText.textColor = UIColor.blueColor()
        }
        UIView.animateWithDuration(2, animations: anim)
    }
}
